tests/integration/conftest.py

```diff
@@ -0,0 +1,210 @@
+
+            # make any internal named volumes unique to the netbox version
+            if "volumes" in new_services[new_service_name]:
+                new_volumes = []
+                for volume_config in new_services[new_service_name]["volumes"]:
+                    source = volume_config.split(":")[0]
+                    if "/" in source:
+                        new_volumes.append(volume_config)
+                    else:
+                        new_volumes.append(
+                            "pytest_pynetbox_"
+                            + netbox_integration_version["docker_tag"]
+                            + "_"
+                            + volume_config
+                        )
+                new_services[new_service_name]["volumes"] = new_volumes
+
+        # replace the servives config with the renamed versions
+        compose_data["services"] = new_services
+
+        # prepend local volume names
+        new_volumes = {}
+        for volume_name, volume_config in compose_data["volumes"].items():
+            new_volumes[
+                "pytest_pynetbox_"
+                + netbox_integration_version["docker_tag"]
+                + "_"
+                + volume_name
+            ] = volume_config
+        compose_data["volumes"] = new_volumes
+
+        compose_output_fpath = os.path.join(
+            netbox_docker_repo_fpath,
+            "docker-compose_%s.yml" % netbox_integration_version["tag"],
+        )
+        with open(compose_output_fpath, "w") as fdesc:
+            fdesc.write(yaml.dump(compose_data))
+
+        compose_files.append(compose_output_fpath)
+
+    return compose_files
+
+
+def is_responsive(url):
+    try:
+        response = requests.get(url)
+        if response.status_code == 200:
+            return True
+    except ConnectionError:
+        return False
+
+
+def id_netbox_service(fixture_value):
+    """Create and ID representation for a netbox service fixture param.
+
+    Returns:
+        str: Identifiable representation of the service, as best we can
+
+    """
+    return "netbox %s" % fixture_value["tag"]
+
+
+@pytest.fixture(
+    scope="session", params=netbox_integration_versions(), ids=id_netbox_service
+)
+def netbox_service(docker_ip, docker_services, request):
+    """Ensure that HTTP service is up and responsive."""
+    netbox_integration_version = request.param
+
+    # `port_for` takes a container port and returns the corresponding host port
+    port = docker_services.port_for(
+        "netbox_%s_nginx" % netbox_integration_version["docker_tag"], 8080
+    )
+    url = "http://{}:{}".format(docker_ip, port)
+    docker_services.wait_until_responsive(
+        timeout=300.0, pause=1, check=lambda: is_responsive(url)
+    )
+    return {
+        "url": url,
+        "netbox_version": netbox_integration_version,
+        "api": pynetbox.api(url, token="0123456789abcdef0123456789abcdef01234567"),
+    }
```

Comment:
I don't love the structure of this being a dict.. it would be much easier to reference an `.api` attribute than the current `["api"]` dict item but that may be just me. It doesn't even really need to contain all this data in here for most test, but it's good to know the netbox version for some tests I think in order to skip some tests on incompatible versions. Ideas welcomed.